SU2_CFD/include/numerics/scalar/scalar_convection.hpp

```diff
@@ -130,9 +130,14 @@ class CUpwScalar : public CNumerics {
         q_ij += 0.5 * (V_i[iDim + idx.Velocity()] + V_j[iDim + idx.Velocity()]) * Normal[iDim];
       }
     }
-
-    a0 = 0.5 * (q_ij + fabs(q_ij));
-    a1 = 0.5 * (q_ij - fabs(q_ij));
+    
+    if(bounded_scalar){
+      a0 = max(0.0, MassFlux);
+      a1 = min(0.0, MassFlux);
+    }else{
+      a0 = 0.5 * (q_ij + fabs(q_ij));
+      a1 = 0.5 * (q_ij - fabs(q_ij));
+    }
```

Comment:
@bigfooted a0 and a1 should have units of m^3 /s and then derived classes either include density or not (e.g. sst and as respectively). As it is now in this PR the units are not consistent if bounded scalars are used (kg / s).

Reply:
The mass flux is now divided by the density when computing a0 and a1. In the ComputeResidual function, the mass fluxes are then multiplied by the density again, maintaining consistent units.